csharp/src/SeedLang/X/SeedPython.cs

```diff
@@ -22,34 +22,72 @@
-      { SeedPythonParser.COLON, SyntaxType.Symbol },
-      { SeedPythonParser.WHILE, SyntaxType.Keyword },
-      { SeedPythonParser.TRUE, SyntaxType.Keyword },
-      { SeedPythonParser.UNKNOWN_CHAR, SyntaxType.Unknown },
+    // The dictionary that maps from ANTLR4 token types of SeedPython to syntax token types.
+    private readonly Dictionary<int, TokenType> _typeMap = new Dictionary<int, TokenType> {
+      // The keys are ordered by the constant names defined in the ANTLR-generated source
+      // SeedPythonParser.cs. Please keep this dictionary up-to-date once the grammar is updated.
+
+      // Ignored: T__0
+      // Ignored: T__1
+      { SeedPythonParser.IF, TokenType.Keyword },
+      { SeedPythonParser.ELIF, TokenType.Keyword },
+      { SeedPythonParser.ELSE, TokenType.Keyword },
+      { SeedPythonParser.FOR, TokenType.Keyword },
+      { SeedPythonParser.IN, TokenType.Keyword },
+      { SeedPythonParser.WHILE, TokenType.Keyword },
+      { SeedPythonParser.DEF, TokenType.Keyword },
+      { SeedPythonParser.RETURN, TokenType.Keyword },
+      { SeedPythonParser.PASS, TokenType.Keyword },
+      { SeedPythonParser.COLON, TokenType.Symbol },
+      { SeedPythonParser.SEMICOLON, TokenType.Symbol },
+      { SeedPythonParser.TRUE, TokenType.Boolean },
+      { SeedPythonParser.FALSE, TokenType.Boolean },
+      { SeedPythonParser.NONE, TokenType.None },
+      { SeedPythonParser.AND, TokenType.Operator },
+      { SeedPythonParser.OR, TokenType.Operator },
+      { SeedPythonParser.NOT, TokenType.Operator },
+      { SeedPythonParser.EQUAL, TokenType.Operator },
+      { SeedPythonParser.EQ_EQUAL, TokenType.Operator },
+      { SeedPythonParser.NOT_EQUAL, TokenType.Operator },
+      { SeedPythonParser.LESS_EQUAL, TokenType.Operator },
+      { SeedPythonParser.LESS, TokenType.Operator },
+      { SeedPythonParser.GREATER_EQUAL, TokenType.Operator },
+      { SeedPythonParser.GREATER, TokenType.Operator },
+      { SeedPythonParser.ADD, TokenType.Operator },
+      { SeedPythonParser.SUBTRACT, TokenType.Operator },
+      { SeedPythonParser.MULTIPLY, TokenType.Operator },
+      { SeedPythonParser.DIVIDE, TokenType.Operator },
+      { SeedPythonParser.FLOOR_DIVIDE, TokenType.Operator },
+      { SeedPythonParser.POWER, TokenType.Operator },
+      { SeedPythonParser.MODULO, TokenType.Operator },
+      { SeedPythonParser.ADD_ASSIGN, TokenType.Operator },
+      { SeedPythonParser.SUBSTRACT_ASSIGN, TokenType.Operator },
+      { SeedPythonParser.MULTIPLY_ASSIGN, TokenType.Operator },
+      { SeedPythonParser.DIVIDE_ASSIGN, TokenType.Operator },
+      { SeedPythonParser.MODULO_ASSIGN, TokenType.Operator },
+      { SeedPythonParser.OPEN_PAREN, TokenType.OpenParenthesis },
+      { SeedPythonParser.CLOSE_PAREN, TokenType.CloseParenthesis },
+      { SeedPythonParser.OPEN_BRACK, TokenType.OpenBracket },
+      { SeedPythonParser.CLOSE_BRACK, TokenType.CloseBracket },
+      // Ignore: OPEN_BRACE

```

Comment:
Braces will be supported in SeedPython to initialize dictionary in the future. Shall we also add them in?

Reply:
Done. Added them.